File: src/netconsole/models/api/wireless_dashboard.py
from __future__ import annotations

from pydantic import Field

from netconsole.models.api.ac_management import AcApDTO, AcManagementSummaryDTO
from netconsole.models.api.ac_mesh_link import AcMeshLinkRecordDTO, AcMeshLinkSummaryDTO
from netconsole.models.api.agent import AgentDTO
from netconsole.models.api.common import ApiModel
from netconsole.models.api.job_center import JobCenterTaskDTO
from netconsole.models.api.mesh_analysis import MeshAnalysisSessionDTO, MeshAnalysisSummaryDTO
from netconsole.models.api.online_mr import OnlineMrSessionSummaryDTO
from netconsole.models.api.train_communication import TrainCommunicationRowDTO, TrainCommunicationSummaryDTO


class WirelessDashboardAlertDTO(ApiModel):
    id: str
    severity: str
    category: str
    title: str
    message: str
    entity_id: str = ""
    detail_path: str = ""
    updated_at: str = ""


class WirelessDashboardFreshnessItemDTO(ApiModel):
    source: str
    label: str
    status: str = "no_data"
    updated_at: str = ""
    age_seconds: int | None = None
    message: str = ""


class WirelessDashboardSummaryDTO(ApiModel):
    site_id: str
    site_name: str = ""
    line_name: str = ""
    ap_total: int = 0
    online_aps: int = 0
    offline_aps: int = 0
    unauthenticated_aps: int = 0
    optical_anomalies: int = 0
    registered_trains: int = 0
    registered_mrs: int = 0
    online_mrs: int = 0
    offline_mrs: int = 0
    stale_mrs: int = 0
    active_online_mr_sessions: int = 0
    agent_total: int = 0
    online_agents: int = 0
    running_tasks: int = 0
    mesh_analysis_sessions: int = 0
    alert_total: int = 0
    critical_alerts: int = 0
    warning_alerts: int = 0
    updated_at: str = ""
    data_version: str = ""
    cached: bool = False


class WirelessDashboardInfrastructureDTO(ApiModel):
    ac: AcManagementSummaryDTO
    mesh_link: AcMeshLinkSummaryDTO
    optical_anomalies: list[AcApDTO] = Field(default_factory=list)
    current_links: list[AcMeshLinkRecordDTO] = Field(default_factory=list)


class WirelessDashboardTrainsDTO(ApiModel):
    summary: TrainCommunicationSummaryDTO
    items: list[TrainCommunicationRowDTO] = Field(default_factory=list)
    total: int = 0


class WirelessDashboardAlertsDTO(ApiModel):
    items: list[WirelessDashboardAlertDTO] = Field(default_factory=list)
    total: int = 0
    critical: int = 0
    warning: int = 0


class WirelessDashboardFreshnessDTO(ApiModel):
    items: list[WirelessDashboardFreshnessItemDTO] = Field(default_factory=list)


class WirelessDashboardRecentOperationsDTO(ApiModel):
    tasks: list[JobCenterTaskDTO] = Field(default_factory=list)
    sessions: list[OnlineMrSessionSummaryDTO] = Field(default_factory=list)


class WirelessDashboardAnalysisDTO(ApiModel):
    summary: MeshAnalysisSummaryDTO
    sessions: list[MeshAnalysisSessionDTO] = Field(default_factory=list)


class WirelessDashboardAgentsDTO(ApiModel):
    items: list[AgentDTO] = Field(default_factory=list)
    total: int = 0
    online: int = 0
    offline: int = 0
    unknown: int = 0


class WirelessDashboardDTO(ApiModel):
    summary: WirelessDashboardSummaryDTO
    infrastructure: WirelessDashboardInfrastructureDTO
    trains: WirelessDashboardTrainsDTO
    alerts: WirelessDashboardAlertsDTO
    freshness: WirelessDashboardFreshnessDTO
    recent_operations: WirelessDashboardRecentOperationsDTO
    analysis: WirelessDashboardAnalysisDTO
    agents: WirelessDashboardAgentsDTO


__all__ = [name for name in globals() if name.startswith("WirelessDashboard")]

File: apps/web/src/types/wirelessDashboard.ts
import type { AcAp, AcManagementSummary } from './acManagement'
import type { AcMeshLinkRecord, AcMeshLinkSummary } from './acMeshLink'
import type { AgentItem } from './agent'
import type { MeshAnalysisSession, MeshAnalysisSummary } from './meshAnalysis'
import type { OnlineMrSessionSummary } from './onlineMr'
import type { TrainCommunicationRow, TrainCommunicationSummary } from './trainCommunication'

export interface WirelessDashboardAlert {
  id: string; severity: string; category: string; title: string; message: string
  entity_id: string; detail_path: string; updated_at: string
}
export interface WirelessDashboardFreshnessItem {
  source: string; label: string; status: string; updated_at: string; age_seconds: number | null; message: string
}
export interface WirelessDashboardTask {
  id: string; type: string; name: string; status: string; progress: number; owner: string; executor: string
  device_name: string; mr_name: string; updated_time: string; error_summary: string; has_warning: boolean
}
export interface WirelessDashboardSummary {
  site_id: string; site_name: string; line_name: string
  ap_total: number; online_aps: number; offline_aps: number; unauthenticated_aps: number; optical_anomalies: number
  registered_trains: number; registered_mrs: number; online_mrs: number; offline_mrs: number; stale_mrs: number
  active_online_mr_sessions: number; agent_total: number; online_agents: number; running_tasks: number
  mesh_analysis_sessions: number; alert_total: number; critical_alerts: number; warning_alerts: number
  updated_at: string; data_version: string; cached: boolean
}
export interface WirelessDashboardInfrastructure {
  ac: AcManagementSummary; mesh_link: AcMeshLinkSummary; optical_anomalies: AcAp[]; current_links: AcMeshLinkRecord[]
}
export interface WirelessDashboardTrains { summary: TrainCommunicationSummary; items: TrainCommunicationRow[]; total: number }
export interface WirelessDashboardAlerts { items: WirelessDashboardAlert[]; total: number; critical: number; warning: number }
export interface WirelessDashboardFreshness { items: WirelessDashboardFreshnessItem[] }
export interface WirelessDashboardRecentOperations { tasks: WirelessDashboardTask[]; sessions: OnlineMrSessionSummary[] }
export interface WirelessDashboardAnalysis { summary: MeshAnalysisSummary; sessions: MeshAnalysisSession[] }
export interface WirelessDashboardAgents { items: AgentItem[]; total: number; online: number; offline: number; unknown: number }
export interface WirelessDashboard {
  summary: WirelessDashboardSummary; infrastructure: WirelessDashboardInfrastructure; trains: WirelessDashboardTrains
  alerts: WirelessDashboardAlerts; freshness: WirelessDashboardFreshness
  recent_operations: WirelessDashboardRecentOperations; analysis: WirelessDashboardAnalysis; agents: WirelessDashboardAgents
}
